src/web/components/ParticipantRequests/ParticipantApprovalForm.tsx

```diff
@@ -0,0 +1,110 @@
+import { ChangeEvent, useMemo, useState } from 'react';
+import { FormProvider, SubmitHandler, useForm } from 'react-hook-form';
+
+import { ParticipantTypeDTO } from '../../../api/entities/ParticipantType';
+import { ParticipantRequestDTO } from '../../../api/routers/participantsRouter';
+import { SiteDTO } from '../../../api/services/adminServiceClient';
+import { ParticipantApprovalFormDetails } from '../../services/participant';
+import { useSiteList } from '../../services/site';
+import { Form } from '../Core/Form';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { Input } from '../Input/Input';
+import { TextInput } from '../Input/TextInput';
+import { SearchBarContainer, SearchBarInput, SearchBarResults } from '../Search/SearchBar';
+
+import './ParticipantApprovalForm.scss';
+
+type ParticipantApprovalFormProps = {
+  onApprove: (formData: ParticipantApprovalFormDetails) => Promise<void>;
+  participant: ParticipantRequestDTO;
+  participantTypes: ParticipantTypeDTO[];
+};
+function ParticipantApprovalForm({
+  onApprove,
+  participant,
+  participantTypes,
+}: ParticipantApprovalFormProps) {
+  const { sites } = useSiteList();
+  const [searchText, setSearchText] = useState(participant.name);
+  const [selectedSite, setSelectedSite] = useState<SiteDTO>();
+
+  const formMethods = useForm<ParticipantApprovalFormDetails>({
+    defaultValues: { name: participant.name, types: participant.types?.map((t) => t.id) },
+  });
+  const { register, handleSubmit, setValue } = formMethods;
+
+  const onSubmit = async (data: ParticipantApprovalFormDetails) => {
+    await onApprove(data);
+  };
+
+  const onSiteClick = (site: SiteDTO) => {
+    setValue('siteId', site.id);
+    setSelectedSite(site);
+  };
+
+  const onSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    // This feature still under development :)
+    // setSearchText(event.target.value);
```

Comment:
I'll follow up with another PR to make search actually work. I'm keen to merge this one before it gets any bigger!
This is fine to merge as-is - no search but you can use the control.